pkg/operator/vspherecontroller/vspherecontroller.go

```diff
@@ -390,7 +392,7 @@ func (c *VSphereController) updateConditions(ctx context.Context, name string, l
 	if lastCheckResult.BlockUpgrade {
 		blockUpgradeMessage := fmt.Sprintf("Marking cluster un-upgradeable because %s", lastCheckResult.Reason)
 		klog.Warningf(blockUpgradeMessage)
-		c.eventRecorder.Warningf(string(lastCheckResult.CheckStatus), "Marking cluster un-upgradeable because %s", lastCheckResult.Reason)
+		c.meteredEventEmitter.Warn(lastCheckResult)
```

Comment:
IMO it would be enough to emit the event only when `allowUpgradeCond` actually changes the existing condition. Would it be better to pass it here (and `Get` it in the caller) instead of the whole `warningEventEmitter`?

Reply:
okay done. PTAL.